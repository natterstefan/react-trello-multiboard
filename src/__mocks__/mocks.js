import { pick } from 'lodash'

// This file contains mocked responses from the data/trello API.
export const mockExampleBoardConfig = {
  board: 'hello-world',
  estimates_with_round_brackets: true,
  estimates_with_square_brackets: true,
  lists: /#upcoming/,
  shortcut: 'hw',
}

export const mockEampleUser1 = {
  avatarHash: '123456789',
  fullName: 'Example Username',
  id: 'member-1',
  initials: 'EU',
  username: 'exampleusername',
}

export const mockEampleUser2 = {
  avatarHash: '987654321',
  fullName: 'Another User',
  id: 'member-2',
  initials: 'AU',
  username: 'anotheruser',
}

export const mockEampleUser3 = {
  avatarHash: null,
  fullName: 'Third User',
  id: 'member-3',
  initials: 'TU',
  username: 'thirduser',
}

export const mockExampleMemberResponse = username => {
  switch (username) {
    case 'anotheruser':
      return mockEampleUser2
    case 'thirduser':
      return mockEampleUser3
    default:
      return mockEampleUser1
  }
}

export const mockExampleBoardResponse = { name: 'hello-world', id: 'board-1' }

// most likely it should be an array with two lists (one matching and one not)
export const mockExampleListResponse = {
  id: 'list-123',
  idBoard: 'board-1',
  name: 'Test List Name #upcoming',
}

export const mockExampleCardsResponse = {
  id: 'card-1',
  name: 'Test Card Name',
  idBoard: 'board-1',
  idMembers: [mockEampleUser1.id, mockEampleUser2.id],
  idList: 'list-1',
  shortUrl: 'https://trello.com/c/abcdefgh',
  members: [
    pick(mockEampleUser1, ['id', 'username', 'fullName']),
    pick(mockEampleUser2, ['id', 'username', 'fullName']),
  ],
}
