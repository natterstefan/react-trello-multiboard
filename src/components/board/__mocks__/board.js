import {
  mockExampleUser1,
  mockExampleUser2,
  mockExampleBoardResponse,
  mockExampleListResponse,
} from '../../../__mocks__/mocks'

export const props = {
  board: {
    id: 'board-1',
  },
}

export const storeStateMock = {
  boards: {
    data: [
      {
        board: mockExampleBoardResponse,
      },
      {
        board: {
          id: 'board-2',
          name: 'another-example-board',
        },
      },
    ],
  },
  lists: {
    'board-1': {
      isLoading: false,
      data: [mockExampleListResponse],
      error: null,
    },
  },
  info: {
    version: 1,
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [mockExampleUser1, mockExampleUser2],
  },
}
