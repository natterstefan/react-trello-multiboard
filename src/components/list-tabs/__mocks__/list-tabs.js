export const storeStateMock = {
  app: {
    config: {
      lists: ['list-1', 'list-2'],
    },
    listToggle: {
      toggleList: 'list-1',
    },
  },
}
