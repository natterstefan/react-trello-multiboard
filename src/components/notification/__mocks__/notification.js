export const props = { message: 'some notification' }

export const storeStateMock = {
  boards: {
    isLoading: false,
    error: null,
  },
  lists: {
    'board-1': {
      isLoading: false,
      error: null,
    },
  },
  members: {
    isLoading: false,
    error: null,
  },
}

export const storeWithBoardError = {
  boards: {
    isLoading: false,
    error: {
      status: 401,
      responseText: 'example response',
    },
  },
}

export const storeWithAnyError = {
  user: {
    error: { message: 'some error user message ' },
  },
}
