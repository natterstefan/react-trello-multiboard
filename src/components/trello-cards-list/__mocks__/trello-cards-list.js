import {
  mockExampleCardsResponse,
  mockExampleBoardConfig,
  mockExampleListResponse,
} from '../../../__mocks__/mocks'

export const props = {
  list: mockExampleListResponse,
  config: mockExampleBoardConfig,
}

export const storeStateMock = {
  cards: {
    'list-1': {
      data: [
        {
          card: mockExampleCardsResponse,
          config: mockExampleBoardConfig,
        },
      ],
      isLoading: true,
      error: null,
    },
  },
}
