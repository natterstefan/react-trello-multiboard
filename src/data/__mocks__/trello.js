// JEST - How to mock modules
//
// either we mock it in each test seperately
// - >>> https://github.com/facebook/jest/issues/2582#issuecomment-346886018
// - https://stackoverflow.com/a/40465435/1238150
//
// or provide a solution for every test in the __mocks__
// - https://facebook.github.io/jest/docs/en/manual-mocks.html
import {
  mockExampleMemberResponse,
  mockExampleBoardResponse,
  mockExampleListResponse,
  mockExampleCardsResponse,
} from '../../__mocks__/mocks'

// NOTE: use mockResolvedValue instead
export const authenticateUser = jest.fn().mockResolvedValue({})
export const getBoard = jest.fn().mockResolvedValue(mockExampleBoardResponse)
export const getMeBoards = jest.fn().mockResolvedValue([mockExampleBoardResponse])
export const getLists = jest.fn().mockResolvedValue([mockExampleListResponse])
export const getCards = jest.fn().mockResolvedValue([mockExampleCardsResponse])
export const getMember = jest.fn(username => Promise.resolve(mockExampleMemberResponse(username)))
export const isTrelloAvailable = jest.fn().mockReturnValue(true)

export default {
  authenticateUser,
  getBoard,
  getMeBoards,
  getLists,
  getCards,
  getMember,
  isTrelloAvailable,
}
