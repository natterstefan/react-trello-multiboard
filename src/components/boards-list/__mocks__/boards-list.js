import { mockExampleUser1, mockExampleBoardResponse } from '../../../__mocks__/mocks'

export const storeStateMock = {
  boards: {
    data: [{ board: mockExampleBoardResponse }],
    error: null,
    isLoading: true,
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: { members: [mockExampleUser1] },
}
