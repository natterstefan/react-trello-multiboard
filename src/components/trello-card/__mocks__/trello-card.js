import { mockEampleUser1, mockEampleUser2 } from '../../../__mocks__/mocks'

export const props = {
  addEstimations: jest.fn(),
  boardName: 'example-board',
  card: {
    badges: {
      description: false,
      comments: 1,
      attachments: 1,
      checkItems: 2,
      checkItemsChecked: 1,
    },
    id: 'id-1',
    idBoard: 'board-1',
    idMembers: ['member-1', 'member-2'],
    members: [mockEampleUser1, mockEampleUser2],
    name: 'example-name',
    shortUrl: 'https://trello.com/id-1',
  },
  config: {
    estimates_with_round_brackets: false,
    estimates_with_square_brackets: false,
  },
  isHidden: false,
  listName: 'example-list',
}

export const storeStateMock = {
  boards: {
    data: [
      {
        board: {
          // will be added to the TrelloCard (because it matches the idBoards above)
          id: 'board-1',
          name: 'example-board',
        },
      },
      {
        board: {
          id: 'board-2',
          name: 'another-example-board',
        },
      },
    ],
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [],
  },
}

export const storeStateMockPreferred = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMember = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-1',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMemberInvalid = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-3',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}
