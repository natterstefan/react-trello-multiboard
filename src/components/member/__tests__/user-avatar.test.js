import React from 'react'
import { shallow } from 'enzyme'
import Avatar from '@material-ui/core/Avatar'

import { mockEampleUser1, mockEampleUser3 } from '../../../__mocks__/mocks'

import UserAvatar from '../user-avatar'

const props = {
  member: {
    ...mockEampleUser1,
    avatarImg: 'http://example.com/avatar.jpg',
  },
}

describe('Component/UserAvatar', () => {
  test('should render without throwing an error', () => {
    expect(shallow(<UserAvatar {...props} />).dive()).toMatchSnapshot()
  })

  test('should render an user w/o an avatar without throwing an error', () => {
    expect(shallow(<UserAvatar member={mockEampleUser3} />).dive()).toMatchSnapshot()
  })

  test('should render an Avatar', () => {
    const wrapper = shallow(<UserAvatar {...props} />).dive()
    expect(wrapper.find(Avatar).length).toEqual(1)
  })
})
