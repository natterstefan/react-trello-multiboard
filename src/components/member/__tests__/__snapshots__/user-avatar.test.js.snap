// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/UserAvatar should render an user w/o an avatar without throwing an error 1`] = `
<div>
  <WithStyles(Avatar)
    className="UserAvatar-initalAvatar-1"
  >
    TU
  </WithStyles(Avatar)>
</div>
`;

exports[`Component/UserAvatar should render without throwing an error 1`] = `
<div>
  <WithStyles(Avatar)
    src="http://example.com/avatar.jpg"
  />
</div>
`;
